src/accounts/avalanche.ts

```diff
@@ -17,19 +16,18 @@ import verifyAvalanche from "../utils/signature/verifyAvalanche";
  * AvalancheAccount implements the Account class for the Avalanche protocol.
  * It is used to represent an Avalanche account when publishing a message on the Aleph network.
  */
-export class AvalancheAccount extends ECIESAccount {
-    private signer?: KeyPair | EVMKeyPair;
-    private provider?: BaseProviderWallet;
-    constructor(signerOrProvider: KeyPair | EVMKeyPair | BaseProviderWallet, address: string, publicKey?: string) {
+export class AvalancheAccount extends EVMAccount {
+    public override readonly wallet?: JsonRPCWallet;
+    public readonly keyPair?: KeyPair | EVMKeyPair;
+    constructor(signerOrWallet: KeyPair | EVMKeyPair | JsonRPCWallet, address: string, publicKey?: string) {
         super(address, publicKey);
-        if (signerOrProvider instanceof KeyPair || signerOrProvider instanceof EVMKeyPair)
-            this.signer = signerOrProvider;
-        if (signerOrProvider instanceof BaseProviderWallet) this.provider = signerOrProvider;
+        if (signerOrWallet instanceof KeyPair || signerOrWallet instanceof EVMKeyPair) this.keyPair = signerOrWallet;
+        if (signerOrWallet instanceof JsonRPCWallet) this.wallet = signerOrWallet;
     }
 
     override GetChain(): Chain {
-        if (this.signer) return Chain.AVAX;
-        if (this.provider) return Chain.AVAX;
+        if (this.keyPair) return Chain.AVAX;
```

Comment:
Why put here this if, if that always return AVAX as Chain. Maybe if you want to make these checks before, will be better to unify it:

```suggestion
        if (this.keyPair && this.wallet) return Chain.AVAX;
```

Reply:
Idk, it was already like this before I took that on \o/
But yeah, will do